sdk/tests/core/test_az_hfsm_pipeline.c

```diff
@@ -0,0 +1,376 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include "az_test_definitions.h"
+#include <azure/core/internal/az_hfsm_pipeline_internal.h>
+#include <azure/core/internal/az_hfsm_internal.h>
+#include <azure/core/internal/az_precondition_internal.h>
+#include <azure/core/az_result.h>
+
+#include <setjmp.h>
+#include <stdarg.h>
+#include <stddef.h>
+
+#include <cmocka.h>
+
+static az_hfsm_pipeline az_hfsm_pipeline_test;
+
+static az_hfsm_policy P01; // Outbound
+static az_hfsm_policy P02;
+static az_hfsm_policy P03; // Inbound
+
+typedef enum
+{
+  P_OUTBOUND_0 = _az_HFSM_MAKE_EVENT(_az_FACILITY_HFSM, 5),
```

Comment:
What does P and S mean? I think the code could be a bit more verbose to clarify the intent.

Reply:
Changed P and S as needed to clear it up :)